samples/dotnet/kernel-syntax-examples/Example35_DIContainer.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Connectors.AI.OpenAI.TextCompletion;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Services;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.TemplateEngine;
+using RepoUtils;
+
+/**
+ * The following examples show how to use SK SDK in applications using DI/IoC containers.
+ */
+public static class Example35_DIContainer
+{
+    public static async Task RunAsync()
+    {
+        await UseKernelInDIPowerAppAsync();
+
+        await UseKernelInDIPowerApp_AdvancedScenarioAsync();
+    }
+
+    /// <summary>
+    /// This example shows how to registered Kernel in DI container without registering it's dependencies and using KernelBuilder instead.
+    /// </summary>
+    private static async Task UseKernelInDIPowerAppAsync()
+    {
+        //Bootstrapping code that initializes the modules, components, and classes that applications use.
+        //For regular .NET applications, the bootstrapping code usually resides either in the Main method or very close to it.
+        //In ASP.NET Core applications, the bootstrapping code is typically located in the ConfigureServices method of the Startup class.
+
+        //Registering Kernel dependencies
+        var collection = new ServiceCollection();
+        collection.AddTransient<ILogger>((_) => ConsoleLogger.Log);
+
+        //Registering Kernel
+        collection.AddTransient<IKernel>((serviceProvider) =>
+        {
+            return Kernel.Builder
+            .WithLogger(serviceProvider.GetRequiredService<ILogger>())
+            .WithOpenAITextCompletionService("text-davinci-002", Env.Var("OPENAI_API_KEY"))
+            .Build();
+        });
+
+        //Registering class that uses Kernel to execute a skill
+        collection.AddTransient<KernelClient>();
+
+        //Creating a service provider for resolving registered services
+        var serviceProvider = collection.BuildServiceProvider();
+
+        //If an application follows DI guidelines, the following line is unnecessary because DI will inject an instance of the KernelClient class to a class that references it.
+        //DI container guidelines - https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#recommendations
```

Comment:
```suggestion
        //If an application follows DI guidelines, the following line is unnecessary because DI will inject an instance 
        //of the KernelClient class to a class that references it.
        //DI container guidelines: 
        // https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#recommendations
```